### DISCUSSION TITLE
How to run the web UI only using CPU and without an NVIDIA graphic card

### DISCUSSION THREAD

**Discussion by alessandro-violante-dsgroup** [oobabooga/text-generation-webui#4033] [Q&A]
Hi,

I'm new to Code Llama and I'm trying to use the web UI with the model "codellama_CodeLlama-7b-Instruct-hf" I have found on Hugging Face.

I've followed these steps:

1. I've created a Conda environment (after having installed Anaconda)
2. I've activated the environment
3. I've installed the CPU only version of PyTorch for Windows
4. I've installed the web UI
5. I've downloaded the model from Hugging Face and I've put it in the models folder
6. I've started the web UI
7. I've tried to load the model using the "Load" button

After having clicked on "Load", after some time, on the command prompt I get the error visible in the attached print screen:

<img width="929" alt="llama-execution-nvidia-error" src="https://github.com/oobabooga/text-generation-webui/assets/143533030/52901753-22b1-4515-9cb1-02040777d845">

I would like to use only my CPU just to make some tests, as I don't have an NVIDIA graphic card.

If this could help in any way, when I clone the repository https://github.com/oobabooga/text-generation-webui, I get this error, although the web UI can be started:

_ERROR: auto_gptq-0.4.2+cu117-cp310-cp310-win_amd64.whl is not a supported wheel on this platform._

I've also tried running this command, but when I try to load the model, I still have the same error:

`pip install -r requirements_nocuda.txt`

What should I do in order to use only my CPU and to be able to make a test? Thank you so much!

**Comment by lucasuniverse** [comment]
Conseguiu resolver seu problema? Utilizando apenas a CPU?  Se sim! Poderia informar o que foi feito?

**Comment by TheLounger** [comment]
Use a model and loader that are primarily designed for CPU, such as [codellama-7b-instruct.Q6_K.gguf](https://huggingface.co/TheBloke/CodeLlama-7B-Instruct-GGUF/blob/main/codellama-7b-instruct.Q6_K.gguf).
Use these settings under the Model tab:
- **Model**: The GGUF model you downloaded
- **Model loader**: llama.cpp
- **n-gpu-layers**: 0
- **threads**: the amount of physical cores on your CPU

Then press Load.

Edit: Oof, necropost 😑
